### ISSUE TITLE
Kubernetes python API error handling

### ISSUE THREAD

**Issue by cmonfort** [scality/metalk8s#1046]
In `metalk8s/salt/_modules/metalk8s_kubernetes.py`, errors when calling Kubernetes API are - mostly - handled this way:
```
    try:
        api_instance = kubernetes.client.CoreV1Api()
        api_response = api_instance.list_node()

        # ... do something with response ...

    except (ApiException, HTTPError) as exc:
        if isinstance(exc, ApiException) and exc.status == 404:
            return None
        else:
            log.exception('Exception when calling CoreV1Api->list_node')
            raise CommandExecutionError(exc)
``` 
So, basically the only error handled is a HTTP 404, with a big assumption it is returned when Kubernetes does not find the object requested... IMHO this is wrong, and we should get a 404 when we call an API that does not exist (so it should be considered a fatal error), and by experiencing with different calls to Kubernetes API it looks like when object is not found a valid response is sent (aka. no HTTP error), but the `response.items` array is just empty.

**Comment by gdemonet** [comment]
As mentioned in [this comment](https://github.com/scality/metalk8s/issues/1047#issuecomment-487935829), this is not debt per-se, as we did not introduce it. It's an enhancement. And I'm removing the milestone as well.

**Comment by TeddyAndrieux** [comment]
Fixed by #1972
